Mathlib/NumberTheory/Cyclotomic/PrimitiveRoots.lean

```diff
@@ -189,6 +189,101 @@ theorem finrank (hirr : Irreducible (cyclotomic n K)) : finrank K L = (n : ℕ).
     (zeta_spec n K L).minpoly_eq_cyclotomic_of_irreducible hirr, natDegree_cyclotomic]
 #align is_cyclotomic_extension.finrank IsCyclotomicExtension.finrank
 
+variable {L} in
+/-- If `L` contains both a primitive `p`-th root of unity and `q`-th root of unity, and
+`Irreducible (cyclotomic (lcm p q) K)` (in particular for `K = ℚ`), then the `finrank K L` is at
+least `(lcm p q).totient`. -/
+theorem _root_.IsPrimitiveRoot.lcm_totient_le_finrank [FiniteDimensional K L] {p q : ℕ} {x y : L}
+    (hx : IsPrimitiveRoot x p) (hy : IsPrimitiveRoot y q)
+    (hirr : Irreducible (cyclotomic (Nat.lcm p q) K)) :
+    (Nat.lcm p q).totient ≤ FiniteDimensional.finrank K L := by
+  rcases Nat.eq_zero_or_pos p with (rfl | hppos)
+  · simp
+  rcases Nat.eq_zero_or_pos q with (rfl | hqpos)
+  · simp
+  let k := PNat.lcm ⟨p, hppos⟩ ⟨q, hqpos⟩
+  obtain ⟨xu, rfl⟩ := hx.isUnit hppos
+  let yu := (hy.isUnit hqpos).unit
+  have hxmem : xu ∈ rootsOfUnity k L :=  by
+    rw [mem_rootsOfUnity, ← Units.val_eq_one, Units.val_pow_eq_pow_val]
+    exact (hx.pow_eq_one_iff_dvd _).2 (Nat.dvd_lcm_left _ _)
+  have hymem : yu ∈ rootsOfUnity k L := by
+    rw [mem_rootsOfUnity, ← Units.val_eq_one, Units.val_pow_eq_pow_val, IsUnit.unit_spec]
+    exact (hy.pow_eq_one_iff_dvd _).2 (Nat.dvd_lcm_right _ _)
+  have hxuord : orderOf (⟨xu, hxmem⟩ : rootsOfUnity k L) = p := by
+    rw [← orderOf_injective (rootsOfUnity k L).subtype Subtype.coe_injective,
+      Subgroup.coeSubtype, Subgroup.coe_mk, ← orderOf_units]
+    exact hx.eq_orderOf.symm
+  have hyuord : orderOf (⟨yu, hymem⟩ : rootsOfUnity k L) = q := by
+    rw [← orderOf_injective (rootsOfUnity k L).subtype Subtype.coe_injective,
+      Subgroup.coeSubtype, Subgroup.coe_mk, ← orderOf_units, IsUnit.unit_spec]
+    exact hy.eq_orderOf.symm
```

Comment:
Can you use [rootsOfUnity_le_of_dvd](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/RootsOfUnity/Basic.html#rootsOfUnity_le_of_dvd) to simplify this proof?

Reply:
Not that much (because of something annoying about `PNat`, you probably don't want to know), but a little golf was possible, thanks!